lib/executor.js

```diff
@@ -87,6 +88,10 @@ function assertLoopBackVersion(app) {
   }
 }
 
+function setEnv(app, instructions) {
+  app.set('env', instructions.env || 'development');
+}
+
```

Comment:
> would you have some insights to why the value of instructions.env becomes undefined in test cases in executor.test.js script , so that's why I set the default value to development here.

Many tests use hand-written instructions created by function `someInstructions`. You should extend that function to provide default `env` when it was not passed in. See https://github.com/strongloop/loopback-boot/blob/088ca864c25e5f10d1c2277e943dd786394be986/test/executor.test.js#L845-L863

> Also it is worth mentioning in this test script it expects test as opposed to development for the this test line on Jenkins!

When not set to a different value, `app.get('env')` returns `NODE_ENV` or `development`. Unit-tests are run with `NODE_ENV=test` on our CI.

I think you need to change `someInstructions` to set `env` to `process.env.NODE_ENV || 'development'`.

----

Actually, forget about what I said above. It will be best to preserve backwards compatibility here and change `setEnv` to a no-op when `instructions.env` is not set.

```js
function setEnv(app, instructions) {
  var env = instructions.env;
  if (env === undefined) return;
  app.set('env', env);
}
```

Then you should add one more test to verify that `options.env` is propagated to `app.get('env')`, if you haven't done so already.